### PR TITLE
Fix index.d.ts

### PR DESCRIPTION
**Pull request by kamektx**
Current version's index.d.ts is broken so you can't get type information of `FastifyInstance.io`.
I fixed it. Please merge and release new version to npm.

### CODE REVIEW

**Comment by alemagio** [comment]
Hi @kamektx , thanks for your PR, would you mind fixing the test too?

File: index.d.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import { FastifyPlugin } from 'fastify'
-import { ServerOptions } from 'socket.io'
+import { FastifyPluginAsync } from 'fastify'
+import * as SocketIO from 'socket.io'
```
Comment (alemagio) [inline index.d.ts]:
```suggestion
import { ServerOptions } from 'socket.io'
```

File: test/index.test-d.ts [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
+      pingInterval: 10000,
+      pingTimeout: 5000,
+      cookie: false
+    })
+  } catch (err) {
+    console.error(err)
+  }
 }
+
+await test()
```
Comment (alemagio) [inline test/index.test-d.ts]: there is no need of wrapping it in an async function since the test library supports top level await

**Review by alemagio — CHANGES_REQUESTED** [review]
@kamektx also you can test on your local machine by running `npm run test` and the when it passes you can push

**Comment by kamektx** [comment]
It seems that the test itself was broken, so I fixed it too, because the new index.d.ts works well when `npm i fastify-socket.io` -> Rewrite index.d.ts in another project.

**Comment by alemagio** [comment]
> It seems that test itself is broken, so I fixed it too, because the new index.d.ts works well when `npm i fastify-socket.io` -> Rewrite index.d.ts in another project.

Yes, since you changed the types the test for the types needs to be changed accordingly.
If you have any question just let me know 😃

**Comment by kamektx** [comment]
Is this OK?

**Comment by alemagio** [comment]
LGTM

**Comment by kamektx** [comment]
Thank you!

**Comment by alemagio** [comment]
Thank you man 🎉

**Comment by kamektx** [comment]
Also I noticed that in `tsconfig.json`, `fastify-tsconfig.json` file is missing.
Could you add it in the repository?

**Comment by alemagio** [comment]
> Also I noticed that in `tsconfig.json`, `fastify-tsconfig.json` file is missing.
> Could you add it in the repository?

What do you mean?

**Comment by kamektx** [comment]
It seems to be my vscode's problem... 
Sorry for annoying.

**Comment by alemagio** [comment]
> It seems to be my vscode's problem...
> Sorry for annoying.

No problem man, thanks for your help and also if there is anything else feel free to open an issue.

**Pull request merged**